src/debug/ee/wks/CMakeLists.txt

```diff
@@ -12,24 +12,44 @@ endif (CLR_CMAKE_PLATFORM_ARCH_I386)
 
 set(ASM_FILE ${CORDBEE_DIR}/${ARCH_SOURCES_DIR}/dbghelpers.asm)
 # asm files require preprocessing using cl.exe on arm64
-  if(CLR_CMAKE_PLATFORM_ARCH_ARM64)
-      get_filename_component(name ${ASM_FILE} NAME_WE)
-      set(ASM_PREPROCESSED_FILE ${CMAKE_CURRENT_BINARY_DIR}/${name}.asm)
-      preprocess_def_file(${ASM_FILE} ${ASM_PREPROCESSED_FILE})
-      set(CORDBEE_SOURCES_WKS_PREPROCESSED_ASM  ${ASM_PREPROCESSED_FILE})
-      set_property(SOURCE ${CORDBEE_SOURCES_WKS_PREPROCESSED_ASM} PROPERTY COMPILE_DEFINITIONS ${ASM_DEFINITIONS})
-      set_property(SOURCE ${CORDBEE_SOURCES_WKS_PREPROCESSED_ASM} PROPERTY COMPILE_DEFINITIONS ${ASM_OPTIONS})
-      add_library_clr(cordbee_wks ${CORDBEE_SOURCES_WKS} ${CORDBEE_SOURCES_WKS_PREPROCESSED_ASM})
-  else ()
-
-      # Need to compile asm file using custom command as include directories are not provided to asm compiler
-      add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/dbghelpers.obj
-                           COMMAND ${CMAKE_ASM_MASM_COMPILER} ${ASM_INCLUDE_DIRECTORIES} ${ASM_DEFINITIONS} ${ASM_OPTIONS} /Fo${CMAKE_CURRENT_BINARY_DIR}/dbghelpers.obj /Ta${ASM_FILE}
-                           DEPENDS ${ASM_FILE}
-                           COMMENT "Compiling dbghelpers.asm")
-
-      add_library_clr(cordbee_wks ${CORDBEE_SOURCES_WKS} ${CMAKE_CURRENT_BINARY_DIR}/dbghelpers.obj)
-  endif(CLR_CMAKE_PLATFORM_ARCH_ARM64)
+if(CLR_CMAKE_PLATFORM_ARCH_ARM64)
+    get_filename_component(name ${ASM_FILE} NAME_WE)
+    set(ASM_PREPROCESSED_FILE ${CMAKE_CURRENT_BINARY_DIR}/${name}.asm)
+    preprocess_def_file(${ASM_FILE} ${ASM_PREPROCESSED_FILE})
+    set(CORDBEE_SOURCES_WKS_PREPROCESSED_ASM  ${ASM_PREPROCESSED_FILE})
+    set_property(SOURCE ${CORDBEE_SOURCES_WKS_PREPROCESSED_ASM} PROPERTY COMPILE_DEFINITIONS ${ASM_DEFINITIONS})
+    set_property(SOURCE ${CORDBEE_SOURCES_WKS_PREPROCESSED_ASM} PROPERTY COMPILE_DEFINITIONS ${ASM_OPTIONS})
+    add_library_clr(cordbee_wks ${CORDBEE_SOURCES_WKS} ${CORDBEE_SOURCES_WKS_PREPROCESSED_ASM})
+elseif(CLR_CMAKE_PLATFORM_ARCH_ARM)
+    
+    # On Arm32 for Windows, use C++ compiler to process the .asm since it includes C-style headers.
+    set(DBGHELPERS_ASM $ENV{__IntermediatesDir}/dbghelpers.asm)
```

Comment:
you can use ${CMAKE_CURRENT_BINARY_DIR} instead of $ENV{__IntermediatesDir}. ${CMAKE_CURRENT_BINARY_DIR} points to bin\obj\src\debug\ee\wks\<flavor> dir here.